### PR TITLE
Add documentation to LinearBoundary stages

### LINKED ISSUE #413: [DOC]: Update LinearBoundaryStage Documentation
### How would you describe the priority of this documentation request

Medium

### Please provide a link or source to the relevant docs

https://docs.nvidia.com/morpheus/_modules/morpheus.stages.boundary.linear_boundary_stage.html

### Describe the problems in the documentation

The linear boundary Ingress/Egress stages do not have the needed documentation for each stage. Missing information:

- [x] General description of what the stage does and how to use it
- [x] Example using the stage
- [x] Documentation of the input parameters
- [x] Documentation of the member variables and methods

### (Optional) Propose a correction

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct
- [X] I have searched the [open documentation issues](https://github.com/nv-morpheus/Morpheus/issues?q=is%3Aopen+is%3Aissue+label%3Adoc) and have found no duplicates for this bug report

### PR DESCRIPTION
**Pull request by drobison00**
Resolves #413

### CODE REVIEW

File: morpheus/stages/boundary/linear_boundary_stage.py [hunk 1]
```diff
@@ -68,11 +80,23 @@ def _build_single(self, builder: srf.Builder, input_stream: StreamPair) -> Strea
...
-
+    boundary_port_id : `str`
+        String corresponding to the name of the LinearBoundaryEgressStage's egress port, use to identify and connect
+        with the appropriate egress.
+    data_type : `object`
+        Data type that this Stage will accept, which will correspond to some existing egress output.
+
+    Example
+    -------
+        boundary_ingress = LinearBoundaryIngressStage(config, "my_boundary_port", int)
```
Comment (mdemoret-nv) [inline morpheus/stages/boundary/linear_boundary_stage.py:99]: Same here. Remove this example in favor of one on `LinearPipeline.add_segment_boundary()`

**Review by mdemoret-nv — CHANGES_REQUESTED** [review]
Looks good. I think we need to add a docstring to `LinearPipeline.add_segment_boundary()` as well since thats the interface most users will use to interact with the segments at first.

File: morpheus/stages/boundary/linear_boundary_stage.py [hunk 2]
```diff
@@ -27,11 +27,23 @@
...
-
+    boundary_port_id : `str`
+        String indicating the name of the egress port associated with the LinearBoundaryEgressStage; allowing it to be
+        paired with the corresponding ingress port.
+    data_type : `python data type instance`
+        Data type that this Stage will accept and then output to its egress port.
+
+    Example
+    -------
+        boundary_egress = LinearBoundaryEgressStage(config, "my_boundary_port", int)
```
[the remark this answers is not in the document]
Comment (drobison00) [reply morpheus/stages/boundary/linear_boundary_stage.py:46]: I'll leave the basic usage example here, and add your suggestion to the LinearPipeline .

**Comment by drobison00** [comment]
@mdemoret-nv Updated based on your most recent comments.

File: morpheus/pipeline/linear_pipeline.py [hunk 3]
```diff
@@ -103,6 +103,35 @@ def add_stage(self, stage: _pipeline.SinglePortStage):
...
+
+            # Add a segment boundary \n
+            # [Current Segment] - [Egress Boundary] ---- [Ingress Boundary] - [Next Segment]
+
+            pipe.add_segment_boundary(MessageMeta)
+
+            # Add a sink in Segment #2\n
+            pipe.add_stage(WriteToFileStage(config, filename=out_file, overwrite=False))
+
+            pipe.run()
```
Comment (mdemoret-nv) [inline morpheus/pipeline/linear_pipeline.py:133]:
```suggestion
            # Create a config and pipeline
            config = Config()
            pipe = LinearPipeline(config)

            # Add a source in Segment #1
            pipe.set_source(FileSourceStage(config, filename=val_file_name, iterative=False))

            # Add a segment boundary
            # [Current Segment] - [Egress Boundary] ---- [Ingress Boundary] - [Next Segment]
            pipe.add_segment_boundary(MessageMeta)

            # Add a sink in Segment #2
            pipe.add_stage(WriteToFileStage(config, filename=out_file, overwrite=False))

            pipe.run()
```
Remove unnecessary `\n`

**Review by mdemoret-nv — APPROVED** [review]
Just one fix then it looks good.

**Comment by mdemoret-nv** [comment]
@gpucibot merge

**Pull request merged**